src/components/SearchPanel/Search/SearchControls.tsx

```diff
@@ -2,18 +2,19 @@ import { useContext, type ReactNode } from 'react';
 import { PokemonListContext } from '../../../types/contexts';
 
 export function SearchControls(): ReactNode {
-  const { query, loading, updateContext } = useContext(PokemonListContext);
+  const { query, loadingButtonSearch, updateContext } =
```

Comment:
you still use loading flag from context
it does not make sense because you can use `useGetPokemonByPageQuery` here and get info about loading status from this query